Apsitvarkom.UnitTests/Api/Controllers/CleaningEventControllerTests.cs

```diff
@@ -210,4 +210,113 @@ public async Task GetById_RepositoryThrows_Status500InternalServerErrorReturned(
         Assert.That(result.StatusCode, Is.EqualTo(StatusCodes.Status500InternalServerError));
     }
     #endregion
+
+    #region Delete tests
+    [Test]
+    public async Task Delete_RepositoryGetsAndDeletesOnce_NoContentResultReturned()
+    {
+        var identifyRequest = new ObjectIdentifyRequest
+        {
+            Id = Guid.NewGuid()
+        };
+
+        _repository.Setup(r => r.ExistsByPropertyAsync(It.IsAny<Expression<Func<CleaningEvent, bool>>>())).ReturnsAsync(true);
+
+        var actionResult = await _controller.Delete(identifyRequest);
+
+        _repository.Verify(r => r.DeleteAsync(It.IsAny<CleaningEvent>()), Times.Once);
+        _repository.Verify(r => r.ExistsByPropertyAsync(It.IsAny<Expression<Func<CleaningEvent, bool>>>()), Times.Once);
+
+        Assert.That(actionResult, Is.TypeOf<NoContentResult>());
+        var result = actionResult as NoContentResult;
+
+        Assert.That(result, Is.Not.Null);
+        Assert.That(result.StatusCode, Is.EqualTo(StatusCodes.Status204NoContent));
+    }
+
+    [Test]
+    public async Task Delete_RepositoryReturnsNull_NotFoundActionResultReturned()
+    {
+        var identifyRequest = new ObjectIdentifyRequest
+        {
+            Id = Guid.NewGuid()
+        };
+
+        _repository.Setup(r => r.ExistsByPropertyAsync(It.IsAny<Expression<Func<CleaningEvent, bool>>>()))
+            .ReturnsAsync(false);
+
+        var actionResult = await _controller.Delete(identifyRequest);
+
+        _repository.Verify(r => r.DeleteAsync(It.IsAny<CleaningEvent>()), Times.Never);
+        _repository.Verify(r => r.ExistsByPropertyAsync(It.IsAny<Expression<Func<CleaningEvent, bool>>>()), Times.Once);
+
+        Assert.That(actionResult, Is.TypeOf<NotFoundObjectResult>());
+        var result = actionResult as NotFoundObjectResult;
+
+        Assert.That(result, Is.Not.Null);
+        Assert.That(result.StatusCode, Is.EqualTo(StatusCodes.Status404NotFound));
+        Assert.That(result.Value, Is.Not.Null.And.Not.Empty);
+    }
+
+    [Test]
+    public async Task Delete_NullIdEntered_ValidationResultsInBadRequestResponseReturned()
+    {
+        var identifyRequest = new ObjectIdentifyRequest
+        {
+            Id = null
+        };
+
+        var actionResult = await _controller.Delete(identifyRequest);
+
+        _repository.Verify(r => r.DeleteAsync(It.IsAny<CleaningEvent>()), Times.Never);
+
+        Assert.That(actionResult, Is.TypeOf<BadRequestObjectResult>());
+        var result = actionResult as BadRequestObjectResult;
+
+        Assert.That(result, Is.Not.Null);
+        Assert.That(result.StatusCode, Is.EqualTo(StatusCodes.Status400BadRequest));
+        Assert.That(result.Value, Is.Not.Null.And.Not.Empty);
+    }
+
+    [Test]
+    public async Task Delete_RepositoryThrowsAcquiringPollutedLocation_Status500InternalServerErrorReturned()
```

Comment:
some names use `PollutedLocation`, however we are now testing `CleaningEvent`s here. let's adjust that

Reply:
Whoops